forms-flow-web/src/components/Form/Item/Edit.js

```diff
@@ -788,19 +865,79 @@ const Edit = React.memo(() => {
       />
 
       <ExportModal
-       showExportModal={selectedAction === EXPORT}
-       onClose={handleCloseSelectedAction}
-       formId={processListData.id}
+        showExportModal={selectedAction === EXPORT}
+        onClose={handleCloseSelectedAction}
+        formId={processListData.id}
+      />
+      <ConfirmModal
+        show={selectedAction === DELETE}
+        title={
+          processListData.id && applicationCount
+            ? t("You Cannot Delete This Form")
+            : t("Are You Sure You Want to Delete This Form?")
+        }
+        message={
+          processListData.id && applicationCount ? (
+            // Case when form has associated submissions and cannot be deleted
+            <div>
+              <p className="fw-bold">
+                {t(
+                  "You may not delete a form that has submissions associated with it."
+                )}
+              </p>
+              <p>
+                {t(
+                  "But you may unpublish it if you wish to not receive any more submissions."
+                )}
+              </p>
+            </div>
+          ) : (
+            // Case when form can be deleted
+            <div>
+              <p className="fw-bold">{t("This action cannot be undone.")}</p>
+            </div>
+          )
+        }
+        primaryBtnAction={handleCloseActionModal}
+        onClose={handleCloseActionModal}
+        secondayBtnAction={ applicationCount ? unPublishActiveForm : deleteModal}
+        primaryBtnText={
+          processListData.id && applicationCount
+            ? t("Keep This Form")
+            : t("No Keep This Form")
+        }
+        secondaryBtnText={
+          processListData.id && applicationCount
+          ? t("Unpublish This Form")
+          : t("Yes Delete the Form")
+        }
+        size="md"
       />
```

Comment:
can we split this to 2 modals ? so that too many ternary operators can be removed .